src/main/java/io/cdap/plugin/gcp/dataplex/common/config/DataplexBaseConfig.java

```diff
@@ -0,0 +1,244 @@
+    if (connection == null || connection.getServiceAccountFilePath() == null &&
+      connection.isServiceAccountFilePath()) {
+      try {
+        ServiceAccountCredentials.getApplicationDefault();
+      } catch (IOException e) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * @return GoogleCredentials based on service account
+   */
+  public GoogleCredentials getCredentials() {
+    GoogleCredentials credentials = null;
+    try {
+      credentials = getCredentialsFromServiceAccount();
+    } catch (IOException e) {
+      LOG.debug("Unable to load service account credentials due to error: {}", e.getMessage());
+    }
+    return credentials;
+  }
+
+  /**
+   * @return Project ID
+   */
+  public String getProject() {
+    if (connection == null) {
+      throw new IllegalArgumentException(
+        "Could not get project information, connection should not be null!");
+    }
+    return connection.getProject();
+  }
+
+  public GCPConnectorConfig getConnection() {
+    return connection;
+  }
+
+  @Nullable
+  public String tryGetProject() {
+    return connection == null ? null : connection.tryGetProject();
+  }
+
+  @Nullable
+  public String getServiceAccount() {
+    return connection == null ? null : connection.getServiceAccount();
+  }
+
+  @Nullable
+  public String getServiceAccountType() {
+    return connection == null ? null : connection.getServiceAccountType();
+  }
+
+  @Nullable
+  public Boolean isServiceAccountFilePath() {
+    return connection == null ? null : connection.isServiceAccountFilePath();
+  }
+
+  /**
+   * Check if service account is a valid one.
+   *
+   * @param failureCollector
+   */
+  public void validateServiceAccount(FailureCollector failureCollector) {
+    if (connection.isServiceAccountJson() || connection.getServiceAccountFilePath() != null) {
+      GoogleCredentials credentials = getCredentials();
+      if (credentials == null) {
+        failureCollector.addFailure(String.format("Unable to load credentials from %s.",
+              connection.isServiceAccountFilePath() ? connection.getServiceAccountFilePath() : "provided JSON key"),
+            "Ensure the service account file is available on the local filesystem.")
+          .withConfigProperty("serviceFilePath")
+          .withConfigProperty("serviceAccountJSON");
+        throw failureCollector.getOrThrowException();
+      }
+    }
+  }
+
+  private GoogleCredentials getCredentialsFromServiceAccount() throws IOException {
+    GoogleCredentials credentials = null;
+    // validate service account
+    if (connection.isServiceAccountJson() || connection.getServiceAccountFilePath() != null) {
+      credentials =
+        GCPUtils.loadServiceAccountCredentials(getServiceAccount(), isServiceAccountFilePath())
+          .createScoped(Lists.newArrayList(DataplexConstants.DATAPLEX_SCOPE));
+    } else {
+      credentials = ServiceAccountCredentials.getApplicationDefault().createScoped(
```

Comment:
Is it necessary to use `ServiceAccountCredentials.getApplicationDefault()` while creating the dataplex service client? In other cloud APIs we don't set anything if credentials are null, they are expected to pick them by default.
Ref - https://github.com/data-integrations/google-cloud/blob/8f25b4807340f7a65d0190dc168d5c9b32e51639/src/main/java/io/cdap/plugin/gcp/common/GCPUtils.java#L186

Reply:
Dataplex APIs doesn't provide any such option, so we are manually selecting this in case of auto-detect.